### PR TITLE
Make systemd configurable

### PR DESCRIPTION
**Pull request by newchain**
I have been using this since @marmarek gave it to me, with a couple of line changes for 3.2.

### CODE REVIEW

**Review by marmarek — CHANGES_REQUESTED** [review]
The code itself is ok, but the missing part is setting HAVE_SYSTEMD anywhere. Can be CFLAGS in daemon/Makefile, for example:
```
ifneq (0,$(USE_SYSTEMD))
CFLAGS += -DHAVE_SYSTEMD
endif
```
(put systemd-related cflags and libs there too).
Then, you can build without systemd by `make USE_SYSTEMD=0`

**Comment by newchain** [comment]
> The code itself is ok, but the missing part is setting HAVE_SYSTEMD anywhere. Can be CFLAGS in daemon/Makefile, for example:

Sorry - you included that part but the hunk got lost while applying.

**Comment by newchain** [comment]
> The code itself is ok, but the missing part is setting HAVE_SYSTEMD anywhere. Can be CFLAGS in daemon/Makefile, for example:

Clarification @marmarek : the updated commit has the Makefile part.

File: daemon/Makefile [hunk 1]
```diff
@@ -2,11 +2,15 @@
...
 EXEEXT =
 
+SYSTEMD ?= 1
+
 ifneq ($(OS),Windows_NT)
 CFLAGS += -fPIC -O2
-CFLAGS += `pkg-config --cflags libsystemd || pkg-config --cflags libsystemd-daemon`
 CFLAGS += `pkg-config --cflags vchan-$(BACKEND_VMM)`
+ifeq ($(SYSTEMD),1)
+CFLAGS += `pkg-config --cflags libsystemd || pkg-config --cflags libsystemd-daemon`
```
Comment (marmarek) [inline daemon/Makefile:11]: `-DHAVE_SYSTEMD` is needed here, in addition to systemd libraries.

**Review by marmarek — CHANGES_REQUESTED** [review]

**Pull request closed without merging**